pkg/auth/authmap_gc.go

```diff
@@ -90,14 +90,14 @@ func (r *authMapGarbageCollector) handleCiliumIdentityEvent(_ context.Context, e
 		if err = r.cleanupMissingIdentities(); err != nil {
 			return fmt.Errorf("failed to cleanup missing identities: %w", err)
 		}
+		r.discoveredCiliumIdentities = nil
```

Comment:
should we remove the key one by one in cleanupMissingIdentities() function instead ?

Reply:
@sayboras i can change this if wanted. and then change the check in the "upsert" whether the map has entries intead of whether its nil?

Reply:
@sayboras would it be ok to bring add this "optimization" in an upcoming PR. so the actual PR can get merged because this one might help in tackling https://github.com/cilium/cilium/pull/26059